packages/design-system/src/components/N8nMenuItem/MenuItem.vue

```diff
@@ -1,7 +1,268 @@
+		:class="{
+			[$style.submenu]: true,
+			[$style.item]: true,
+			[$style.compact]: compact,
+			[$style.active]: mode === 'router' && isItemActive(item)
+		}"
+		:index="item.id"
+		:popper-append-to-body="false"
+		:popper-class="`${$style.submenuPopper} ${popperClass}`"
+	>
+			<template slot="title">
+				<n8n-icon v-if="item.icon" :class="$style.icon" :icon="item.icon" :size="item.customIconSize || 'large'" />
+				<span :class="$style.label">{{ item.label }}</span>
+			</template>
+		<el-menu-item
+			v-for="child in availableChildren"
+			:key="child.id"
+			:id="child.id"
+			:class="{
+				[$style.menuItem]: true,
+				[$style.disableActiveStyle]: !isItemActive(child),
+				[$style.active]: isItemActive(child),
+			}"
+			:index="child.id"
+			@click="onItemClick(child)"
+		>
+			<n8n-icon v-if="child.icon" :class="$style.icon" :icon="child.icon" />
+			<span :class="$style.label">{{ child.label }}</span>
+		</el-menu-item>
+	</el-submenu>
+	<n8n-tooltip v-else placement="right" :content="item.label" :disabled="!compact" :open-delay="tooltipDelay">
+		<el-menu-item
+			:id="item.id"
+			:class="{
+				[$style.menuItem]: true,
+				[$style.item]: true,
+				[$style.disableActiveStyle]: !isItemActive(item),
+				[$style.active]: isItemActive(item),
+				[$style.compact]: compact
+			}"
+			:index="item.id"
+			@click="onItemClick(item)"
+		>
+			<n8n-icon v-if="item.icon" :class="$style.icon" :icon="item.icon" :size="item.customIconSize || 'large'" />
+			<span :class="$style.label">{{ item.label }}</span>
+		</el-menu-item>
+	</n8n-tooltip>
+</template>
+
 <script lang="ts">
+import ElSubmenu from 'element-ui/lib/submenu';
 import ElMenuItem from 'element-ui/lib/menu-item';
+import N8nTooltip from '../N8nTooltip';
+import N8nIcon from '../N8nIcon';
+import { IMenuItem } from '../../types';
+import Vue from 'vue';
+import { Route } from 'vue-router';
+
+export default Vue.extend({
+	name: 'n8n-menu-item',
+	components: {
+		ElSubmenu, // eslint-disable-line @typescript-eslint/no-unsafe-assignment
+		ElMenuItem, // eslint-disable-line @typescript-eslint/no-unsafe-assignment
+		N8nIcon,
+		N8nTooltip,
+	},
+	props: {
+		item: {
+			type: Object as () => IMenuItem,
+			required: true,
+		},
+		compact: {
+			type: Boolean,
+			default: false,
+		},
+		tooltipDelay: {
+			type: Number,
+			default: 300,
+		},
+		popperClass: {
+			type: String,
+			default: '',
+		},
+		mode: {
+			type: String,
+			default: 'router',
+			validator: (value: string): boolean => ['router', 'tabs'].includes(value),
+		},
+		activeTab: {
+			type: String,
+		},
+	},
+	computed: {
+		availableChildren(): IMenuItem[] {
+			return this.item.children ? this.item.children.filter(child => child.available !== false) : [];
+		},
+	},
+	methods: {
+		isItemActive(item: IMenuItem): boolean {
+			const isItemActive = this.isActive(item);
+			const childActive = item.children !== undefined && item.children.find(child => this.isActive(child)) !== undefined;
```

Comment:
I would make every expression evaluate to `boolean` and `Array.find` returns the found element or undefined so maybe better if you check if there is any active item (also variable name could be `hasActiveChild`)
```suggestion
			const hasActiveChild = Array.isArray(item.children) && item.children.some(child => this.isActive(child));
```